source/detail/implementations/stylesheet.hpp

```diff
@@ -426,7 +416,10 @@ struct stylesheet
     {
         format_impl new_format = *pattern;
         new_format.style = style_name;
-
+        if (pattern->references == 0)
+        {
+            *pattern = new_format;
+        }
         return find_or_create(new_format);
```

Comment:
I'm not sure why this is still required when references == 0. Should be investigated as it would cut run time significantly (3-4 operations are often executed on new formats before they are applied)